### PR TITLE
control: add guard rails in flight control paths

### PR DESCRIPTION
**Pull request by vlordier**
This draft isolates a set of control-path robustness changes from the larger hardening branch.

Summary:
- add arithmetic / finite-value guards in Copter, L1, AP_Math, and AP_Motors control paths
- add focused AP_Math regression coverage for invalid accel-limit inputs and extreme gain / denominator edge cases

Why:
- these changes were originally buried inside a much larger branch
- this draft allows review of the control-path behavior changes on their own

Validation:
- branch was split cleanly from upstream/master
- the branch includes focused AP_Math regression test updates for the edge cases covered by this slice
- not fully validated locally as a standalone branch beyond the split and commit flow
- draft only: this slice includes behavior-affecting control changes and needs careful review

### CODE REVIEW

File: libraries/AP_Motors/AP_MotorsMulticopter.cpp [hunk 1]
```diff
@@ -446,6 +446,12 @@ void AP_MotorsMulticopter::Log_Write()
 // in all other states: map [0..1] linearly between pwm_min and pwm_max
 int16_t AP_MotorsMulticopter::output_to_pwm(float actuator)
 {
+    // NaN or Inf in the actuator value (from upstream thrust/mixing
+    // calculations) would produce an undefined int16_t cast.  Clamp to
+    // the safe minimum so the motor outputs a known-safe value.
+    if (!isfinite(actuator)) {
+        actuator = 0.0f;
+    }
```
Comment (IamPete1) [inline libraries/AP_Motors/AP_MotorsMulticopter.cpp:454]: If you have evidence that this can actually happen we should fix that not try and deal with it here.
